packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js

```diff
@@ -41,6 +44,9 @@ const DropdownItem = ({ className, children, isHovered, component: Component, is
       <Component
         {...additionalProps}
         className={css(isDisabled && styles.modifiers.disabled, isHovered && styles.modifiers.hover, className)}
+        onClick={() => {
+          Component === 'button' ? (props.onClick && props.onClick(), onSelect()) : onSelect();
```

Comment:
Nice, def cleaner than my suggestion!  Just a couple more things :)

1.  Now that we're explicitly setting `onClick` handler using a prop, `onClick` should be added to the propTypes, and can be destructured along with the other props that we exclude from `...props`
1.  We should probably pass the onClick `event` object to the callbacks, just in case the consumers want/need it
2.  Seeing an eslint error here
    > [eslint] Expected an assignment or function call and instead saw an expression.  [no-unused-expressions]

    We can do something like the following to fix the error
    ```js
    onClick={event => {
      if (Component === 'button') {
        onClick && onClick(event);
        onSelect(event);
      } else {
        onSelect(event);
      }
    }}
    ```
    That being said, I'm not sure why Travis isn't running `prettier`, or maybe this particular rule has been removed.  Are we not standardizing around `prettier` and `eslint`?  @jschuler